libs/ledger-live-common/src/wallet-api/utils/txTrackingHelper.ts

```diff
@@ -0,0 +1,8 @@
+import { Transaction as EvmTransaction } from "@ledgerhq/coin-evm/types/transaction";
+import { DAPP_SELECTORS } from "@ledgerhq/hw-app-eth";
+
+export const getTxType = (tx: EvmTransaction) => {
+  const txSelector = `0x${tx?.data?.toString("hex").substring(0, 8)}`;
+
+  return Object.keys(DAPP_SELECTORS).includes(txSelector) ? DAPP_SELECTORS[txSelector] : "transfer";
+};
```

Comment:
An idea not using the `Object.keys` and avoiding the potential `0xundefined`
```suggestion
const fallback = "transfer";
export function getTxType(tx: EvmTransaction): string {
  if (!tx || !tx.data) return fallback;

  const txSelector = `0x${tx.data.toString("hex").substring(0, 8)}`;
  const type = DAPP_SELECTORS[txSelector];
  
  if (!type) return fallback;
  return type;
};
```